src/main/java/seedu/duke/AddCommand.java

```diff
@@ -0,0 +1,85 @@
+package seedu.duke;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.GregorianCalendar;
+
+/**
+ * Handles adding tasks to the TaskList.
+ */
+public class AddCommand extends Command {
+    protected static String title;
+    protected static String details;
+
+    /**
+     * Constructor.
+     * @param s String containing Task title.
+     * @param det String containing Task timing.
+     */
+    public AddCommand(String s, String det) {
+        title = s;
+        details = det;
+        type = "add";
+    }
+
+    /**
+     * Executes the AddCommand and adds a corresponding Task to the TaskList.
+     *  @param t TaskList.
+     * @param u Ui.
+     * @param s Storage.
+     * @return a String representing the add message
+     */
+    @Override
+    public String execute(TaskList t, Ui u, Storage s) {
+        Task task = createNewTask();
+        int size = t.list.size();
+        t.list.add(task);
+        assert t.list.size() == size + 1: "Current list size should be one more";
+
+        try {
+            String text = task.getType() + " . 0 . " + task.getFullDescription() + "\n";
+            s.update(true, text, t);
+        } catch (IOException e) {
+            System.out.println("Something went wrong");
+        } finally {
+            return u.addLine(task.toString(), t.list.size());
+        }
+    }
+
+    /**
+     * Creates a new Task according to the user input.
+     * @return a new Task.
+     */
+    public static Task createNewTask() {
+        Task t = null;
+        if (title.equals("todo")) {
```

Comment:
Should all this differentiating types of tasks and splitting be left to the parser instead?